frontend/src/api/client.test.ts

```diff
@@ -0,0 +1,65 @@
+import { AxiosRequestConfig, AxiosResponse } from "axios";
+import { httpClient } from ".";
+
+jest.mock("next/router");
+
+const requestInterceptor = () => {
+  return httpClient.interceptors.request['handlers'][0];
+}
+
+const responseInterceptor = ()=> {
+  return httpClient.interceptors.response['handlers'][0];
+}
+
+const mockRequest: AxiosRequestConfig = {
+  url: undefined,
+  headers: {}
+}
+
+const mockResponse = {
+  response: {
+    status: undefined,
+  }
+}
+
+describe('Client interceptors', () => {
+  describe('Request interceptors', () => {
+    it('should request to an open endpoint without authorization', () => {
+      mockRequest.url = '/users/sign_in';
+      const resolvedReq = requestInterceptor().fulfilled(mockRequest);
+      expect(resolvedReq).toMatchObject(mockRequest);
+      expect('Authorization' in resolvedReq.headers).toBeFalsy();
+    });
+    it('should request closed endpoints with authorization', () => {
+      mockRequest.url = '/users/test';
+      const resolvedReq = requestInterceptor().fulfilled(mockRequest);
+      expect(resolvedReq).toMatchObject(mockRequest);
+      expect('Authorization' in resolvedReq.headers).toBeTruthy();
+    });
+    it('should return error', () => {
```

Comment:
Describe better when error occurred

Reply:
should work without authorization when endpoint is public
should work with authorization when endpoint is private
should fail without authorization when endpoint is private

@mizek1 , I think you should go with something like I described above. This would be similar to what you have done at the bottom specs